presto-common/src/main/java/com/facebook/presto/common/type/LongEnumType.java

```diff
@@ -72,12 +80,15 @@ public String getDisplayName()
     public static class LongEnumMap
     {
         private final Map<String, Long> enumMap;
+        private final Map<Long, String> flippedEnumMap;
 
         @JsonCreator
         public LongEnumMap(@JsonProperty("enumMap") Map<String, Long> enumMap)
         {
             validateEnumMap(enumMap);
             this.enumMap = normalizeEnumMap(enumMap);
+            this.flippedEnumMap = this.enumMap.entrySet().stream()
```

Comment:
If you want to further improve the performance, we can make this map lazily instantiated. Add an `AtomicBoolean` to mark whether the map is instantiated, and create it on first call of `getKeyForValue`. That way, if the query doesn't need it, you don't need to create this.